front/src/applications/stdcmV2/views/StdcmViewV2.tsx

```diff
@@ -1,66 +1,187 @@
 
-  const [currentSimulationInputs, setCurrentSimulationInputs] = useState<
-    StdcmSimulationResult['input'] | undefined
-  >(undefined);
+  const dispatch = useAppDispatch();
+  const { resetStdcmConfig, updateStdcmConfigWithData } =
+    useOsrdConfActions() as StdcmConfSliceActions;
 
-  const [interactedResultsElements, setInteractedResultsElements] = useState(false);
+  const selectedSimulation = simulationsList[selectedSimulationIndex];
+  const isCalculationFailed = !isSuccessful && !isStdcmResultsEmpty;
+  const showResults = !isPending && (showStatusBanner || simulationsList.length > 0);
+
+  const handleRetainSimulation = () => setRetainedSimulationIndex(selectedSimulationIndex);
+
+  const handleSelectSimulation = (index: number) => {
+    if (retainedSimulationIndex === -1) {
+      setSelectedSimulationIndex(index);
+      setShowBtnToLaunchSimulation(false);
+    }
+  };
+
+  const handleStartNewQuery = () => {
+    setSimulationsList([]);
+    setSelectedSimulationIndex(-1);
+    setRetainedSimulationIndex(-1);
+    dispatch(resetStdcmConfig());
+  };
+
+  // reset config data with the selected simulation data
+  useEffect(() => {
+    if (selectedSimulation) {
+      const { departureDate, departureTime, pathSteps, consist } = selectedSimulation.inputs;
+
+      dispatch(
+        updateStdcmConfigWithData({
+          rollingStockID: consist?.tractionEngine?.id,
+          speedLimitByTag: consist?.speedLimitByTag,
+          pathSteps: [...pathSteps],
+          originDate: departureDate,
+          originTime: departureTime,
+        })
+      );
+    }
+  }, [selectedSimulation]);
 
   useEffect(() => {
-    setInteractedResultsElements(false);
+    setShowBtnToLaunchSimulation(!isEqual(currentSimulationInputs, selectedSimulation?.inputs));
   }, [currentSimulationInputs]);
 
+  useEffect(() => {
+    if (isPending) {
+      setShowBtnToLaunchSimulation(false);
+    }
+  }, [isPending]);
+
+  useEffect(() => {
+    if (isRejected && !isStdcmResultsEmpty) {
+      setShowStatusBanner(true);
+    }
+  }, [isRejected, isStdcmResultsEmpty]);
+
+  // handle update simulation list on stdcm success or empty results
+  useEffect(() => {
+    if (currentSimulationInputs) {
+      const newSmulationId = simulationsList.length + 1;
+      if (stdcmV2Results?.stdcmResponse && pathProperties) {
+        setSimulationsList([
+          ...simulationsList,
+          {
+            id: newSmulationId,
+            creationDate: new Date(),
+            inputs: currentSimulationInputs,
+            outputs: {
+              results: stdcmV2Results.stdcmResponse,
+              pathProperties,
+            },
+          },
+        ]);
+        setShowStatusBanner(true);
+      } else if (isStdcmResultsEmpty) {
+        setSimulationsList([
+          ...simulationsList,
+          {
+            id: newSmulationId,
+            creationDate: new Date(),
+            inputs: currentSimulationInputs,
+          },
+        ]);
```

Comment:
https://github.com/user-attachments/assets/7c562080-1823-4759-bc6b-64806c96b14e

Seems like something is wrong when the simulation fails. In the video, I have some simulations in the history, I just changed for a different input (chambery - modane), the simulation fails but when checking the result, I still have the one from the previous simulation.

Moreover, sometimes the simulation result is not displayed in the navigator and when launching another one, 2 simulations are popping together in the nav.

What's stranger is that it seems the proper simulation is saved but it's not properly displayed in the result, see below video : 

https://github.com/user-attachments/assets/d9edca7a-42b5-42b1-8633-3e5344fa4458

Maybe it comes from this useEffect refacto ?